pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -202,12 +203,14 @@ func (d *Deployment) fetchPods(ctx context.Context) error {
 	d.status.changed = false
 	for _, p := range pods {
 		originalPod, ok := d.pods[p.String()]
```

Comment:
Could we rename `ok` to `found` or something?  I was puzzling trying to figure out why the "call" would have failed and whether referencing p.StatusCode could be valid.